### DISCUSSION TITLE
Having Database Lock issues with Azure Mobile Apps v4.2.0

### DISCUSSION THREAD

**Discussion by ShaneVick** [Azure/azure-mobile-apps#648] [Q&A]
We are encountering a database locking issue running the 4.2 version of the Microsoft.Azure.Mobile.Client.SQLiteStore package
running in a Xamarin Forms Project,

Microsoft.WindowsAzure.MobileServices.SQLiteStore.SQLiteException: Error executing SQLite command: 'cannot start a transaction within a transaction'.

Microsoft.WindowsAzure.MobileServices.SQLiteStore.SQLiteException: Error executing SQLite command: 'database is locked'.

When the issue arises, the only way we have found to correct the lock is to kill and restart the app,

Is there a way to release/kill connections when the app enters this state, or a method we should be calling to ensure this doesn't occur,

Any help would be greatly appreciated

**Comment by adrianhall** [comment]
Version 4.2.0 is no longer supported.  Since I'd need to see some code to see what is going on, my only recommendation is to re-produce with v5 of the client.  That's obviously a bigger issue than just a version swap (since your server side has to be updated as well).
